src/ImageSharp.Web/Commands/QueryCollectionRequestParser.cs

```diff
@@ -16,17 +15,15 @@ public sealed class QueryCollectionRequestParser : IRequestParser
         /// <inheritdoc/>
         public CommandCollection ParseRequestCommands(HttpContext context)
         {
-            if (context.Request.Query.Count == 0)
+            IQueryCollection query = context.Request.Query;
+            if (query is null || query.Count == 0)
             {
                 // We return new to ensure the collection is still mutable via events.
                 return new();
             }
 
-            // TODO: Investigate skipping the double allocation here.
-            // In .NET 6 we can directly use the QueryStringEnumerable type and enumerate stright to our command collection
-            Dictionary<string, StringValues> parsed = QueryHelpers.ParseQuery(context.Request.QueryString.ToUriComponent());
             CommandCollection transformed = new();
-            foreach (KeyValuePair<string, StringValues> pair in parsed)
+            foreach (KeyValuePair<string, StringValues> pair in context.Request.Query)
```

Comment:
Can't this also use the captured `query` variable?
```suggestion
            foreach (KeyValuePair<string, StringValues> pair in query)
```

Reply:
Yeah just missed it.